inc/Engine/Optimization/DynamicLists/DataManager.php

```diff
@@ -0,0 +1,136 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\DynamicLists;
+
+class DataManager {
+	/**
+	 * Filesystem instance
+	 *
+	 * @var WP_Filesystem_Direct
+	 */
+	private $filesystem;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param WP_Filesystem_Direct $filesystem Filesystem instance.
+	 */
+	public function __construct( $filesystem = null ) {
+		$this->filesystem = is_null( $filesystem ) ? rocket_direct_filesystem() : $filesystem;
+	}
+
+	/**
+	 * Gets the lists content
+	 *
+	 * @return object
+	 */
+	public function get_lists() {
+		$transient = get_transient( 'wpr_dynamic_lists' );
+
+		if ( false !== $transient ) {
+			return $transient;
+		}
+
+		$json = $this->get_lists_from_file();
+
+		$lists = json_decode( $json );
+
+		if ( empty( $lists ) ) {
+			return '';
+		}
+
+		$this->set_lists_cache( $lists );
+
+		return $lists;
+	}
+
+	/**
+	 * Returns the hash of the current JSON
+	 *
+	 * @return string
+	 */
+	public function get_lists_hash() {
+		return md5( $this->get_lists_from_file() );
+	}
+
+	/**
+	 * Save dynamic lists on file & transient
+	 *
+	 * @param string $content Lists content.
+	 *
+	 * @return boolean
+	 */
+	public function save_dynamic_lists( string $content ) {
+		$result = $this->put_lists_to_file( $content );
+
+		$lists = json_decode( $content );
+
+		$this->set_lists_cache( $lists );
+
+		return $result;
+	}
+
+	/**
+	 * Gets the path to the dynamic lists JSON file
+	 *
+	 * @return string
+	 */
+	private function get_json_filepath(): string {
```

Comment:
@Tabrisrp 
if we are going to use GitHub actions
- we need to change the path to be in the plugin directory
- remove fallback file stuff
- update related tests

Reply:
The dynamic list file in the plugin directory is the fallback one. We're still going to store and update the one in `wp-rocket-config` separately on a weekly basis with the cron.

What we're missing in this PR for now is the GH action